rust/main/utils/run-locally/src/main.rs

```diff
@@ -435,7 +436,7 @@ fn main() -> ExitCode {
     let mut failure_occurred = false;
     let starting_relayer_balance: f64 = agent_balance_sum(9092).unwrap();
 
-    while !SHUTDOWN.load(Ordering::Relaxed) {
+    'main_loop: while !SHUTDOWN.load(Ordering::Relaxed) {
```

Comment:
if you need to use labels, it's a strong indication the code reached a complexity where it needs to be refactored quite a bit

you can use that fact that once `termination_invariants_met` is true, the relayer can be restarted before killing the other processes

hoping lines 439-505 can be simplified quite a lot

Reply:
Simplified lines 439-505 a bit :+1: